src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDeconstructTests.cs

```diff
@@ -1164,13 +1164,96 @@ public void Deconstruct(out int a, out int b)
         }
 
         [Fact]
-        public void NoArgIteratorTypeInAsync()
+        public void Constraints_01()
+        {
+            string source = @"
+unsafe class C
+{
+    public void M()
+    {
+        (int x, var (err1, y)) = (0, new C());
+        (int* err2, var err3) = M2();
+        foreach ((int* err4, var err5) in new[] { M2() })
+        {
+        }
+    }
+
+    public static (int*, int*) M2()
+    {
+        return (default(int*), default(int*));
+    }
+    public void Deconstruct(out int* a, out int b)
+    {
+        a = default(int*);
+        b = 2;
+    }
+
+    public void M3()
+    {
+        (int x, var (err1, y)) = (0, new C());
```

Comment:
I imitate this test: http://source.roslyn.io/#Roslyn.Compilers.CSharp.Emit.UnitTests/CodeGen/CodeGenDeconstructTests.cs,dcc965c2bf962ba7

Should I remove duplicates in that one, too?